### ISSUE TITLE
`DeepEquals`: comparator of `SortedSet`s and `SortedMap`s is not considered, equality of ordered and unordered collections is inconsistent with JDK

### ISSUE THREAD

**Issue by Robyt3** [jdereg/java-util#74]
The comparison of two `SortedSet`s or `SortedMap`s with `deepEquals` does not work as expected, when the `SortedSet`s or `SortedMap`s are not using the same comparator for ordering the elements. If not the same comparators are used, the method returns wrong results, as the iterators `i1` and `i2` used in `compareOrderedCollection` and `compareSortedMap` return elements in a different order.

With the default implementation of `equals`, this is not the case:

```java
SortedMap<String, String> map1 = new TreeMap<>(Comparator.naturalOrder());
map1.put("a", "b");
map1.put("c", "d");
SortedMap<String, String> map2 = new TreeMap<>(Comparator.reverseOrder());
map2.put("a", "b");
map2.put("c", "d");
System.out.println(map1.equals(map2)); // true
```

In the JDK, the `equals` method of `Set`s and `Map`s does not consider the ordering in any case, even for `SortedSet`s and `SortedMap`s. Instead `SortedSet`s are considered to be equal to `Set`s (and respectively for `Map`s) when they contain exactly the same entries:

```java
SortedSet<String> set1 = new TreeSet<>();
Set<String> set2 = new HashSet<>();
System.out.println(set1.equals(set2)); // true
```

Whereas `deepEquals` considers those two `Set`s to be not equal, because they are not both of the type `SortedSet`s.

**Comment by jdereg** [comment]
This has been corrected and will be released on version 2.1.0.

**Comment by jdereg** [comment]
Can you verify DeepEquals.deepEquals() works correctly for you?  The issue was tied to DeepEquals as far as I recall.  I realize you are talking about GraphComparator here.  We will take a look at that as it should have same behavior as DeepEquals.
